src/boost-api/endpoint_handlers.cpp

```diff
@@ -1,20 +1,152 @@
+            std::string db_table = jsonData.as_object().at("db_table").as_string().c_str();
+
+            query_purpose purpose = (db_table == "original") ?
+                                                              query_purpose::insert_into_original_rtree_table :
+                                                              query_purpose::insert_into_simplified_rtree_table;
+
+            trajectory_manager::load_trajectories_into_rtree(purpose);
+        } catch (const std::exception& e) {
+            res.result(status::bad_request);
+            res.set(field::content_type, "text/plain");
+            res.body() = "Failed to read 'db_table' value from JSON: " + std::string(e.what());
+        }
+    }
+
+    void handle_spatial_range_query_on_rtree_table(const request<string_body> &req, response<string_body> &res) {
+        boost::json::value jsonData = get_json_from_request_body(req, res);
+        if (jsonData.is_null())
+            return;
+
+        try {
+            query_purpose purpose;
+
+            if (jsonData.as_object().contains("original")) {
+                purpose = query_purpose::load_original_rtree_into_datastructure;
+            } else if (jsonData.as_object().contains("simplified")) {
+                purpose = query_purpose::load_simplified_rtree_into_datastructure;
+            } else {
+                res.result(status::bad_request);
+                res.set(field::content_type, "text/plain");
+                res.body() = "Invalid JSON format: missing 'original' or 'simplified' key";
+                return;
+            }
+
+            boost::json::object originalData = jsonData.as_object().at("original").as_object();
+
+            double x_low = originalData.at("longitudeRange").as_object().at("min").as_double();
+            double x_high = originalData.at("longitudeRange").as_object().at("max").as_double();
+
+            double y_low = originalData.at("latitudeRange").as_object().at("min").as_double();
+            double y_high = originalData.at("latitudeRange").as_object().at("max").as_double();
+
+            long double t_low = originalData.at("timestampRange").as_object().at("min").as_double();
+            long double t_high = originalData.at("timestampRange").as_object().at("max").as_double();
+
+            spatial_queries::Range_Query::Window window{};
+            window.x_low = x_low;
+            window.x_high = x_high;
+            window.y_low = y_low;
+            window.y_high = y_high;
+            window.t_low = t_low;
+            window.t_high = t_high;
+
+            trajectory_manager::spatial_range_query_on_rtree_table(purpose, window);
+        } catch (const std::exception &e) {
+            res.result(status::bad_request);
+            res.set(field::content_type, "text/plain");
+            res.body() = "Error processing JSON data: " + std::string(e.what());
+        }
+    }
+
+    void handle_reset_all_data(const request<string_body> &req, response<string_body> &res) {
```

Comment:
Dette endpoint skal ikke eksistere, da alle vil kunne reset databasen. Lav i stedet et flag, som kan reset databasen, hvis man vælger at inkludere det. Dette kunne være `--reset` eller `--fresh` osv.

Sørg for at main ser således ud: `int main(int argc, char* argv[]) {...}`. Derefter kan i bruge string comparision på `argv`, som er en liste af arguments. `argc` er count af arguments. Og `argv[0]` er altid programmets  navn. Dvs. hvis man kører `./traceq --reset --bigcock`, så er  `argv[0] = "./traceq"`,  `argv[1] = "--reset"`, og  `argv[2] = "--bigcock"`